core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractConfigBuilder.java

```diff
@@ -119,7 +119,8 @@ public static void withCustomizer(SmallRyeConfigBuilder builder, String customiz
     public static ConfigClass configClass(final String mappingClass, final String prefix) {
         try {
             // To support mappings that are not public
-            return ConfigClass.configClass(Thread.currentThread().getContextClassLoader().loadClass(mappingClass), prefix);
+            Class<?> klass = Thread.currentThread().getContextClassLoader().loadClass(mappingClass);
+            return ConfigClass.configClass(klass, prefix);
```

Comment:
I'm correct that no changes were made here? You just split the call in two lines.

How exactly do we leverage the cache?